packages/fxa-settings/src/components/UnitRowSecondaryEmail/index.tsx

```diff
@@ -1,80 +1,226 @@
+    RESEND_SECONDARY_EMAIL_CODE_MUTATION
+  );
+
+  const alertBarAndContent = () => {
+    if (alertBarRevealed && (data || error)) {
+      return (
+        <AlertBar onDismiss={hideAlertBar}>
+          {data && (
+            <p>
+              Check the inbox for {primaryEmail} to verify your secondary email.
             </p>
-          </Modal>
-        )}
-
-        {alertBarRevealed && (
-          <AlertBar onDismiss={hideAlertBar}>
-            <p>Check the inbox for {primaryEmail} to verify your primary email.</p>
-          </AlertBar>
-        )}
-      </UnitRow>
+          )}
+          {error && <p>Error text TBD. {error.message}</p>}
+        </AlertBar>
+      );
+    }
+    return null;
+  };
+
+  const UnitRowSecondaryEmailContentAndActions = ({
+    secondaryEmailObj,
+    index,
+  }: UnitRowSecondaryEmailContentAndActionsProps) => {
+    const secondaryEmail = secondaryEmailObj.email;
+    const secondaryEmailIsVerified = secondaryEmailObj.verified;
+    const ctaText = secondaryEmailIsVerified ? 'Make primary' : '';
+
+    const resendSecondaryEmailCodeHandler = () => {
+      resendSecondaryEmailCode({
+        variables: { input: { email: secondaryEmail } },
+      });
+      revealAlertBar();
+    };
+
+    return (
+      <>
+        {alertBarAndContent()}
+        <div className="flex unit-row-multi-row">
+          <div className="unit-row-content">
+            <p className="font-bold" data-testid="unit-row-header-value">
+              {secondaryEmail}
+              {!secondaryEmailIsVerified && (
+                <span className="uppercase block text-orange-600 font-bold text-xs">
+                  unverified
+                </span>
+              )}
+            </p>
+            {secondaryEmailIsVerified &&
+              lastVerifiedSecondaryEmailIndex === index && (
+                <VerifiedEmailContent />
+              )}
+            {!secondaryEmailIsVerified && (
+              <p className="text-xs mt-3 text-grey-400">
+                Verification needed.
+                <button
+                  className="link-blue mx-1"
+                  onClick={resendSecondaryEmailCodeHandler}
+                >
+                  Resend verification email
+                </button>
+                if it's not in your email or spam.
+              </p>
+            )}
+          </div>
+          <div className="unit-row-actions">
+            <div>
+              {ctaText && (
+                <button
+                  className="cta-neutral transition-standard"
+                  onClick={() => {
+                    console.log(`make ${secondaryEmail} the primary email!`);
+                  }}
+                >
+                  {ctaText}
+                </button>
+              )}
+            </div>
+          </div>
+        </div>
+      </>
+    );
+  };
+
+  if (!hasAtLeastOneSecondaryEmail) {
+    const route = primaryEmailIsVerified
+      ? '/beta/settings/secondary_email'
```

Comment:
`constantantized` is my new favorite word 😛 They both are set to `const` so I'm not sure I know what you mean - this was a bit funky though.

I also prefer dashes, but using the same URLs as current settings takes a lot of redirects out of the question. There's probably a lot of SUMO links, posts all around the 'net, even bookmarked URLs that we'd want to account for otherwise.

Reply:
> I'm not sure I know what you mean

Sorry, I could have been more clear. With how frequently specific routes will be references (from setting up the router, to `<Link />` usage throughout the app, to testing) I think it would be beneficial to use a global paths object, similar to how content-server uses a [constants file](https://github.com/mozilla/fxa/blob/main/packages/fxa-content-server/app/scripts/lib/constants.js). This would allow IDEs to auto-complete/auto-import paths, and eliminate accidental incorrect paths (e.g. someone uses a dash instead of an underscore).

> I also prefer dashes, but using the same URLs as current settings takes a lot of redirects out of the question. There's probably a lot of SUMO links, posts all around the 'net, even bookmarked URLs that we'd want to account for otherwise.

I think a relatively simple solution would be to 304 Redirect the old URLs to the new (looks like React Router might be able to do something similar with [`push`](https://github.com/ReactTraining/react-router/blob/master/packages/react-router/docs/api/Redirect.md#push-bool)). IMO this feels like one of those things we should try to do while we have the opportunity, before things are set in stone? Then again, it's only underscores 😛

Reply:
Ah I see, that sounds like a good idea to me and should probably be done in #6131 or a follow up!

It might be worth bringing up to Alex/Meridel. I know we briefly talked about it but just agreed to stick with parity stuff for less complications.

Reply:
Sweet!

And Sweet! This is also something that could be done at a later date before launch.